scipy/sparse/linalg/interface.py

```diff
@@ -823,4 +821,5 @@ def aslinearoperator(A):
                                   rmatmat=rmatmat, dtype=dtype)
 
         else:
-            raise TypeError('type not understood')
+            A = np.atleast_2d(np.asarray(A))
```

Comment:
There is no protection against making a `LinearOperator` out of an existing `ndarray` unless it has too many dimensions.

If some sort of protection against making a `LinearOperator` out of objects for which linear operations are not defined, maybe we should not use `np.atleast_2d` here? Or we could add `dtype` restrictions as in the first commit (273c7a64285162c8d04c99233d077f7398f1d3f3)